packages/trace-viewer/src/ui/watchMode.tsx

```diff
@@ -299,6 +308,13 @@ export const TestList: React.FC<{
         <div className='watch-mode-list-item-title'>{treeItem.title}</div>
         <ToolbarButton icon='play' title='Run' onClick={() => runTreeItem(treeItem)} disabled={!!runningState}></ToolbarButton>
         <ToolbarButton icon='go-to-file' title='Open in VS Code' onClick={() => sendMessageNoReply('open', { location: locationToOpen(treeItem) })}></ToolbarButton>
+        <ToolbarButton icon='eye' title='Watch' onClick={() => {
+          if (watchedTreeIds.has(treeItem.id))
+            watchedTreeIds.delete(treeItem.id);
+          else
+            watchedTreeIds.add(treeItem.id);
+          setWatchedTreeIds(watchedTreeIds);
```

Comment:
This line is a no-op, because it's the same set.

Reply:
I'm not sure you followed the code.